### ISSUE TITLE
[Bug or Help] python setup.py giving me an error on line 23

### ISSUE THREAD

**Issue by Goldenfreddy0703** [jjjchens235/bing-rewards#172]
Hey there, im not sure if im doing this correctly but im using a raspberry pi and it has dietpi on it, I installed python 3 and chromium and did the command "python setup.py" but i got an error on it for some odd reason. 

```
dietpi@DietPi:~/bing-rewards $ python setup.py
  File "setup.py", line 23
    print(f"   Hashed: {email}\n")
```
I'm not sure if its a bug or its just me on this one but if you can take a look and tell me what im doing wrong, that would be great.

Thank you

**Comment by jjjchens235** [comment]
full error log please

**Comment by Goldenfreddy0703** [comment]
Oh sorry for asking but how would I send a full error log?

**Comment by jjjchens235** [comment]
```sh
dietpi@DietPi:~/bing-rewards $ python setup.py
  File "setup.py", line 23
    print(f"   Hashed: {email}\n")
```

When you hit the error, I'm assuming that the full error was longer than this. If you have access to that still, please post it (or re-run again)

**Comment by Goldenfreddy0703** [comment]
oh im sorry the error was caused by a syntax error and it says invalid syntax. 

Here is a screenshot of what i have so far:
![image](https://user-images.githubusercontent.com/62456796/154262471-2a985602-40c5-4a8c-a8f9-de12605115f3.png)

**Comment by jjjchens235** [comment]
that log is good, you were just missing that last line as you said.

**Comment by jjjchens235** [comment]
Can you run 
`which python`
on the console?

**Comment by Goldenfreddy0703** [comment]
so far this is what i get when entering these commands: 
![image](https://user-images.githubusercontent.com/62456796/154263491-f4023751-4ec8-4367-8953-15f5ab8c7dc5.png)

**Comment by Goldenfreddy0703** [comment]
I just realized, i do not have an update version of python, that could be the culprit.

**Comment by jjjchens235** [comment]
^exactly

**Comment by Goldenfreddy0703** [comment]
ok well im gonna have to talk to the dietpi developers to update there python i got from there applist, Thank you so much for helping me on this.

Im gonna close this issue but if i do have any problems i will reopen and ask for more assistance. 

Again, Thank you

**Comment by jjjchens235** [comment]
No problem!

**Comment by jjjchens235** [comment]
in your 1st screenshot actually, i seem some references to python 3.7 when you run requirements.txt.

i would try figuring out if your system has it already installed possibly and it's just pointing to the wrong 1.

a good first step would be running the program 1 more time but like so:
`python3 BingRewards.py`

**Comment by Goldenfreddy0703** [comment]
ahh ok i will try that, i was looking at this form and i found out that it had both versions installed: https://dietpi.com/phpbb/viewtopic.php?p=26251#p26251

**Comment by Goldenfreddy0703** [comment]
Hey so it looks like i have python3 and python2 so whenever i tried `python3 setup.py` it worked but does chromium support bing rewards?

**Comment by jjjchens235** [comment]
I think you have to install chromedriver, maybe something like this:
https://stackoverflow.com/a/64979264

**Comment by Goldenfreddy0703** [comment]
ahhh ok thank you

**Comment by Goldenfreddy0703** [comment]
Hey so i installed the chromedriver but im getting this error when i run the script:
```
root@DietPi:~/bing-rewards# python3 BingRewards/BingRewards.py -a
Traceback (most recent call last):
  File "BingRewards/BingRewards.py", line 4, in <module>
    from src.rewards import Rewards
  File "/root/bing-rewards/BingRewards/src/rewards.py", line 310
    if progress := self.__get_search_progress(driver, search_type):
                 ^
SyntaxError: invalid syntax
root@DietPi:~/bing-rewards#

```

**Comment by jjjchens235** [comment]
Ahh.. it's because you're on python 3.7 and this syntax is only supported in 3.8+.

I will revert this code back to support 3.7 in the next release.

In the meantime, you will need to adjust the code like so to be 3.7 compatible:
https://github.com/jjjchens235/bing-rewards/blob/531e22c964bae323dc2dbd61b6b7cee44dc01919/BingRewards/src/rewards.py#L292-L296

**Comment by jjjchens235** [comment]
Better yet, the `dev` branch already has that change. Until master becomes integrated, maybe you can use that instead. The instructions are like so:

```sh
git checkout dev
git pull
python3 BingRewards.py
```

**Comment by Goldenfreddy0703** [comment]
Hey thank you so much, I will try out the dev and see if I can get it working

**Comment by Goldenfreddy0703** [comment]
Hey so i cloned the dev branch, followed the instructions and everything but i got the same error. 
```
root@DietPi:~/bing-rewards# python3 BingRewards/BingRewards.py
Traceback (most recent call last):
  File "BingRewards/BingRewards.py", line 4, in <module>
    from src.rewards import Rewards
  File "/root/bing-rewards/BingRewards/src/rewards.py", line 310
    if progress := self.__get_search_progress(driver, search_type):
                 ^
SyntaxError: invalid syntax

```

**Comment by Goldenfreddy0703** [comment]
Actually i copied the wrong link, let me try again

**Comment by jjjchens235** [comment]
The easy fix is to go into your `config.py` file and remove the telegram values like so:
```py
credentials = dict(
    email = 'your_hashed_email',
    password = 'your_hashed_pw',
    telegram_api_token = '',
    telegram_userid = ''
)
```

I'm curious how you got this error though, hoping you can answer these questions so I can try to debug this better.

- Did you re-run setup.py on the dev branch? 
- In your most recent run of setup.py, what action did you take when prompted for telegram credentials?

**Comment by Goldenfreddy0703** [comment]
1. yes i did re-run the setup.py cause it said email=none even tho i put my email in setup.py
2. i just pressed enter in order to skip them. I just put my email and password.

**Comment by Goldenfreddy0703** [comment]
I think the script is now working as it should now. It did everything i asked it to do and now its all working. Thank you so much for helping me on this issue.

**Comment by jjjchens235** [comment]
Thank you as well! You helped me find & fix some issues!
